app/models/role.rb

```diff
@@ -171,6 +182,7 @@ def clone(role_params = {})
   end
 
   def locked?
+    return false if self.class.modify_locked
```

Comment:
There's a check in https://github.com/theforeman/foreman/pull/4623/files#diff-7863dd934f73db4d813673b9dec6db0aR217 for that already. 

The problem is that if I modify the `not_locked` validation to include `self.class.modify_locked`, it will only prevent name modifications and things like that. In order to prevent adding permissions in addition to all of that, I can either:

* Put it here
or
* Put it in `not_locked` and `Filter#role_not_locked` (at least)

Since (IMO) Filter shouldn't really need to depend on a class variable from Role, I put it here.